packages/angular/src/lib/flag.directive.ts

```diff
@@ -1,39 +1,58 @@
-import {
-  Directive,
-  Input,
-  OnInit,
-  TemplateRef,
-  ViewContainerRef,
-} from "@angular/core";
-
+import { Directive, Input, OnDestroy, OnInit, TemplateRef, ViewContainerRef } from "@angular/core";
+import { Subject, takeUntil, map, distinctUntilChanged, ReplaySubject, switchMap } from "rxjs";
 import { AbbyService } from "./abby.service";
 
 @Directive({
   selector: "[featureFlag]",
 })
-export class AbbyFlag implements OnInit {
-  @Input() featureFlag: string;
+export class AbbyFlag implements OnInit, OnDestroy {
+  @Input()
+  set featureFlag(featureFlag: string) {
+    // ensure featureFlag is a string to quit gracefully
+    if(typeof featureFlag !== 'string') {
+      console.warn(`Expected a string as featureFlag. Got ${featureFlag}`);
+      return;
+    }
+
+    this.currentFlag$.next(featureFlag);
+  }
+
+  private currentFlag$ = new Subject<string>();
+  private destroy$ = new Subject<void>();
 
   constructor(
     private readonly abby: AbbyService,
-
     private _viewContainer: ViewContainerRef,
     private _templateRef: TemplateRef<any>
   ) {}
 
   ngOnInit(): void {
-    let flagName: string;
-    if (this.featureFlag.startsWith("!"))
-      flagName = this.featureFlag.substring(1);
-    else flagName = this.featureFlag;
-    this.abby.getFeatureFlagValue(flagName).subscribe((value) => {
-      this._viewContainer.clear();
-      if (
-        (value && this.featureFlag[0] != "!") ||
-        (this.featureFlag[0] == "!" && !value)
-      ) {
-        this._viewContainer.createEmbeddedView(this._templateRef);
-      }
-    });
+    this.currentFlag$
+      .pipe(
+        switchMap((flagName) => {
+          return this.abby
+            .getFeatureFlagValue(flagName.startsWith("!") ? flagName.slice(1) : flagName)
+            .pipe(
+              map(
+                (value) =>
+                  (value && flagName[0] !== "!") || (flagName[0] === "!" && !value)
+              )
+            );
+        }),
+        distinctUntilChanged(),
+        takeUntil(this.destroy$)
+      )
+      .subscribe((visible) => {
```

Comment:
flags arent really flags anymore, right? these can be ANYthing. so.. this visible not visible thingy is not up-to-date anymore!

guess we need either multiple directives.. and this one is the boolean one.. or there should be another input configuring this thingy here?! :)

Reply:
In the scope of the `abbyFlag` Directive it would be the visibility, wouldn't it?
Feature flags evaluate to booleans and that boolean decides whether the content of this directive should be visible.
Or am I missing something?

Reply:
I have no clue. that just came into my mind seeing this. :shrug: should be well documented then. :)

Reply:
actually. if I think more about it. you dont want this boolean behavior. if its a string, I may want to check if it like `my-desired-value` f.e..
or if its a JSON-object, I may want to check if `myobj.nested.trigger` is true?
or if its a number, I may want to check if its above `100`?

Reply:
so.. maybe just call it "boolean thingy" or write some docs/comments that this is only testing the given value in a boolean manner.

and make it properly in another feature PR :)

Reply:
Oh wait, I forgot feature flags could be more than just simple booleans, so yeah, your comment makes sense.

Reply:
> so.. maybe just call it "boolean thingy" or write some docs/comments that this is only testing the given value in a boolean manner.
> 
> and make it properly in another feature PR :)

Yeah i guess I will document it for now. We gotta make a decision on how we want to implement this, so that it is easy and understandable to use.